libs/api/repository/src/lib/gn4/platform/gn4-platform.service.ts

```diff
@@ -0,0 +1,76 @@
+import { Injectable } from '@angular/core'
+import { Observable, of, switchMap } from 'rxjs'
+import { map, shareReplay, tap } from 'rxjs/operators'
+import {
+  MeApiService,
+  SiteApiService,
+  UsersApiService,
+} from '@geonetwork-ui/data-access/gn4'
+import { PlatformServiceInterface } from '@geonetwork-ui/common/domain/platform.service.interface'
+import {
+  MeUserModel,
+  UserModel,
+} from '@geonetwork-ui/common/domain/model/user/user.model'
+import { Organization } from '@geonetwork-ui/common/domain/model/record'
+import { Gn4PlatformMapper } from './gn4-platform.mapper'
+
+const minApiVersion = '4.2.0'
+@Injectable()
+export class Gn4PlatformService implements PlatformServiceInterface {
+  private me$: Observable<MeUserModel>
+  private users$: Observable<UserModel[]>
+  isAnonymous$: Observable<boolean>
+
+  settings$ = of(true).pipe(
+    switchMap(() => this.siteApiService.getSiteOrPortalDescription()),
+    shareReplay(1)
+  )
+
+  readonly apiVersion$ = this.settings$.pipe(
+    map((info) => info['system/platform/version'] as string),
+    shareReplay(1)
+  )
+
+  readonly isApiCompatible$ = this.apiVersion$.pipe(
+    tap(
+      (version) =>
+        version < minApiVersion &&
```

Comment:
Does that really work for semver? e.g.  `4.10.0` might be considered "lower" than `4.2.0`

Reply:
Yes this is not correct, I will add the test of the version in another PR and keep this one as just refactoring.
I don't want that some heavy work is started and brings conflicts because of this refactoring.